libjob_queue/tests/job_queue_timeout_test.c

```diff
@@ -133,11 +152,14 @@ int main(int argc, char ** argv) {
       util_exit("Job failed to be queued!\n");
     }
 
-    usleep(1 * sec); // 1.0 sec
-    int job_status = job_queue_iget_job_status(queue, 0);
+    usleep(4 * sec); // 4.0 sec - should be large enough so that the queue has the time to restart and the exit the job
+
+    for(int ijob = 0; ijob < number_of_jobs; ++ijob) {
+       if(jobs[ijob]->retry_call_count < 1)
+         util_exit("Job's retry callback should have been called at least once, instead it was never called");
 
-    if (job_status != JOB_QUEUE_IS_KILLED) {
-      util_exit("Job should have been killed, had status %d != %d\n", job_status, JOB_QUEUE_IS_KILLED);
+     if(jobs[ijob]->exit_call_count < 1)
+          util_exit("Job's exit callback should have been called at least once, instead it was never called");
```

Comment:
Try just adding a call to `job_queue_manager_wait( )` and then `job_queue_free( )` here.